Pixel Game/Assets/Scripts/TraderNpc.cs

```diff
@@ -66,4 +67,18 @@ private void OnTriggerExit2D(Collider2D collision)
     }
 
 
+    public void OpenNpcShop()
+    {
+        if (shopScreen.GetComponent<CanvasGroup>().alpha == 1) // If shopscreen is visible
+        {
+            shopScreen.GetComponent<ShopScreen>().CloseShopScreen();
+        }
+        else if (shopScreen.GetComponent<CanvasGroup>().alpha == 0) // If shopscreen is NOT visible
+        { 
+            shopScreen.GetComponent<ShopScreen>().OpenShopScreen();
+        }

```

Comment:
A shorter way of writing would be to just check alpha once and then an empty else:
```c#
if (alpha > 0)
    close shop
else
    open shop
```

Reply:
Nice input!